src/mdsynthesis/data.py

```diff
@@ -0,0 +1,394 @@
+        """Return path to datafile corresponding to given handle.
+
+        :Arguments:
+            *handle*
+                name of dataset whose datafile path to return
+
+        :Returns:
+            *datafile*
+                datafile path; None if does not exist
+            *proxyfile*
+                proxyfile path; None if does not exist
+            *datafiletype*
+                datafile type; either ``persistence.pddatafile`` or
+                ``persistence.npdatafile``
+
+        """
+        datafile = None
+        datafiletype = None
+        for dfiletype in (pddata.pddatafile, npdata.npdatafile,
+                          pydata.pydatafile):
+            dfile = os.path.join(self._tree.abspath, handle, dfiletype)
+            if os.path.exists(dfile):
+                datafile = dfile
+                proxyfile = os.path.join(self._tree.abspath, handle,
+                                         ".{}.proxy".format(dfiletype))
+                datafiletype = dfiletype
+
+        if datafile is None and datafiletype is None:
+            raise KeyError("No data for '{}'".format(handle))
+
+        return (datafile, proxyfile, datafiletype)
+
+    def _read_datafile(func):
+        """Decorator for generating DataFile instance for reading data.
+
+        DataFile instance is generated and mounted at self._datafile. It
+        is then dereferenced after the method call. Since data files can be
+        deleted in the filesystem, this should handle cleanly the scenarios
+        in which data appears, goes missing, etc. while a Treant is loaded.
+
+        .. note:: Methods wrapped with this decorator need to have *handle* as
+                  the first argument.
+
+        """
+
+        @wraps(func)
+        def inner(self, handle, *args, **kwargs):
+            filename, proxy, filetype = self._get_datafile(handle)
+
+            if filename:
+                self._datafile = DataFile(
+                    os.path.join(self._tree.abspath, handle),
+                    datafiletype=filetype)
+                try:
+                    out = func(self, handle, *args, **kwargs)
+                finally:
+                    del self._datafile
+            else:
+                out = None
+
+            return out
+
+        return inner
+
+    def _write_datafile(func):
+        """Decorator for generating DataFile instance for writing data.
+
+        DataFile instance is generated and mounted at self._datafile. It
+        is then dereferenced after the method call. Since data files can be
+        deleted in the filesystem, this should handle cleanly the scenarios
+        in which data appears, goes missing, etc. while a Treant is loaded.
+
+        .. note:: Methods wrapped with this decorator need to have *handle* as
+                  the first argument.
+
+        """
+
+        @wraps(func)
+        def inner(self, handle, *args, **kwargs):
+            dirname = os.path.join(self.treant.abspath, handle)
```

Comment:
@dotsdl  This change enables to run the tests without an explicit error being thrown. But now there is an infinite recursion happening saving a dataframe.

Reply:
This was due to a missing pytables package. Still a weird error at least to me.